src/Model/PayPalPurchaseUnit.php

```diff
@@ -18,80 +18,29 @@
 
 class PayPalPurchaseUnit
 {
-    /** @var string */
-    private $referenceId;
-
-    /** @var string */
-    private $invoiceNumber;
-
-    /** @var string */
-    private $currencyCode;
-
-    /** @var int */
-    private $totalAmount;
-
-    /** @var int */
-    private $shippingValue;
-
-    /** @var float */
-    private $itemTotalValue;
-
-    /** @var float */
-    private $taxTotalValue;
-
-    /** @var int */
-    private $discountValue;
-
-    /** @var string */
-    private $softDescriptor;
-
-    /** @var string */
-    private $merchantId;
-
-    /** @var ?AddressInterface */
-    private $shippingAddress;
-
-    /** @var bool */
-    private $shippingRequired;
-
-    /** @var array */
-    private $items;
-
     public function __construct(
-        string $referenceId,
-        string $invoiceNumber,
-        string $currencyCode,
-        int $totalAmount,
-        int $shippingValue,
-        float $itemTotalValue,
-        float $taxTotalValue,
-        int $discountValue,
-        string $merchantId,
-        array $items,
-        bool $shippingRequired,
-        ?AddressInterface $shippingAddress = null,
-        string $softDescriptor = 'Sylius PayPal Payment',
+        private readonly string $referenceId,
+        private readonly string $invoiceId,
+        private readonly string $currencyCode,
+        private readonly int $totalAmount,
+        private readonly int $shippingValue,
+        private readonly float $itemTotalValue,
+        private readonly float $taxTotalValue,
+        private readonly int $discountValue,
+        private readonly int $shippingDiscountValue,
```

Comment:
Add this new argument as the optional one at the end and add the note to the UPGRADE file